lib/index.ts

```diff
@@ -276,10 +276,14 @@ export default class Backburner {
       }
     }
 
-    if (length === 1) {
-      return method();
-    } else if (length === 2) {
-      return method.call(target);
+    let onError = getOnError(this.options);
+
+    if (onError) {
+      try {
```

Comment:
method call is wrapped in `try catch` in `run` already, we can use this technique https://github.com/bekzod/rsvp.js/blob/6b75926e62810a20e3951a4db08de25b3f8bb802/lib/rsvp/-internal.js#L40 this should perform better in old browsers I think